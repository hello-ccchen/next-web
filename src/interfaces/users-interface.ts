export interface IUser {
  id: number;
  email: string;
  first_name: string;
  last_name: string;
  avatar: string;
}

export interface IUserAPIResponse {
  page: number;
  per_page: number;
  total: number;
  total_pages: number;
  data: IUser[];
  support?: { url: string; text: string };
}

export interface IUserState {
  userList: IUser[];
  isLoading: boolean;
  error: string | undefined;
}
