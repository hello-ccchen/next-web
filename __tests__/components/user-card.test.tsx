/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react";
import UserCard from "@/components/user-card";
import { IUser } from "@/interfaces/users-interface";

it("renders UserCard component unchanged", () => {
  const mockUser: IUser = {
    id: 0,
    email: "abcd@email.com",
    first_name: "foo",
    last_name: "bar",
    avatar: "http:image/test/abcde.jpg",
  };
  const { container } = render(<UserCard {...mockUser} />);
  expect(container).toMatchSnapshot();
});
