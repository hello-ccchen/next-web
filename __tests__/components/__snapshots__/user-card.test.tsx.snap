// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UserCard component unchanged 1`] = `
<div>
  <div
    class="d-flex flex-wrap justify-content-center align-items-center my-auto"
  >
    <img
      alt="avatar"
      class="m-2 rounded-circle"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="150"
      src="/_next/image?url=http%3Aimage%2Ftest%2Fabcde.jpg&w=384&q=75"
      srcset="/_next/image?url=http%3Aimage%2Ftest%2Fabcde.jpg&w=256&q=75 1x, /_next/image?url=http%3Aimage%2Ftest%2Fabcde.jpg&w=384&q=75 2x"
      style="color: transparent;"
      width="150"
    />
    <div
      class="d-flex flex-column align-items-center w-100"
    >
      <h6>
        foo bar
      </h6>
      <div
        class="d-flex align-items-center"
      >
        <span>
          ****@email.com
        </span>
        <img
          alt="mask"
          class="ms-2"
          data-nimg="1"
          decoding="async"
          height="16"
          loading="lazy"
          src="/eye.svg"
          style="color: transparent; cursor: pointer;"
          width="16"
        />
      </div>
    </div>
  </div>
</div>
`;
